### PR TITLE
[core] Support `@mui/utils` v6

### LINKED ISSUE #14866: [core] Unstable_ClassNameGenerator (mui v6) didn't transform DataGrid classes
### Steps to reproduce

Link to live example mui v5 (working): [codesandbox](https://codesandbox.io/p/sandbox/basiccolumnsgrid-material-5-demo-wscdpc)

Steps:
1. Check classes of Button and DataGrid
2. Everything transformed (has `toto-` prefix)

Link to live example mui v6 (broken): [codesandbox](https://codesandbox.io/p/sandbox/elegant-tree-xspyf2)

Steps:
1. Check classes of button and datagrid
2. DataGrid classes not transformed, only Button class transformed

### Expected behavior

DataGrid classes must be transformed

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
Happens in any browser

System:
    OS: Linux 5.15 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
  Binaries:
    Node: 20.17.0 - ~/.nvm/versions/node/v20.17.0/bin/node
    npm: 10.8.2 - ~/.nvm/versions/node/v20.17.0/bin/npm
    pnpm: Not Found
  Browsers:
    Chrome: Not Found
```
</details>

**Search keywords**: datagrid, Unstable_ClassNameGenerator, generator, className
**Order ID**: 80084

**Search keywords**:

### PR DESCRIPTION
**Pull request by siriwatknp**
> Feel free to change the PR title, this fix applies to all mui-x products not just DataGrid

closes #14866

## Root cause

From https://codesandbox.io/p/sandbox/elegant-tree-xspyf2?file=%2Finitialize.ts%3A5%2C1,

It's the inconsistency of `@mui/utils` version.
- `@mui/material` => `@mui/utils` v6
- `@mui-x/data-grid` => `@mui/utils` v5

That's why configuring the classNameGenerator from `@mui/material` (using `@mui/utils` v6) does not affect DataGrid which is using `@mui/utils` v5.

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Review by michelengelen — COMMENTED** [review]
LGTM, but I would like to wait for one more review on this, since I am not aware of all implications this might have.

**Review by LukasTy — APPROVED** [review]
I've [forked](https://codesandbox.io/p/sandbox/basiccolumnsgrid-material-6-demo-forked-sjj889) the mentioned CodeSandbox and installed a build from this branch.
Class name generation works as expected. 👍 

Thank you for taking care of it! 🙏

File: packages/x-internals/package.json
Comment (michelengelen) [reply packages/x-internals/package.json:45]: Nice spot!

**Pull request merged**
